File: programmers-coding-test/C++/Level1/콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int collatz(long n, int count) {
    if (count == 500) return -1;
    if (n == 1) return count;
    return n % 2 ? collatz(n * 3 + 1, count + 1) : collatz(n / 2, count + 1);
}

int solution(int num) {
    return collatz(num, 0);
}

File: programmers-coding-test/Python/Level1/콜라츠 추측.py
def solution(num):
    work = 0
    while num != 1 and work < 500:
        if num % 2 == 0:
            num /= 2
        else:
            num = num * 3 + 1
        work += 1
    return work if work < 500 else -1
